cmd/notation/blob/verify.go

```diff
@@ -0,0 +1,169 @@
+				return errors.New("--media-type is set but with empty value")
+			}
+			opts.Common.Parse(cmd)
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runVerify(cmd, opts)
+		},
+	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	command.Flags().StringVar(&opts.signaturePath, "signature", "", "filepath of the signature to be verified")
+	command.Flags().StringArrayVar(&opts.pluginConfig, "plugin-config", nil, "{key}={value} pairs that are passed as it is to a plugin, if the verification is associated with a verification plugin, refer plugin documentation to set appropriate values")
+	command.Flags().StringVar(&opts.blobMediaType, "media-type", "", "media type of the blob to verify")
+	command.Flags().StringVar(&opts.policyStatementName, "policy-name", "", "policy name to verify against. If not provided, the global policy is used if exists")
+	cmd.SetPflagUserMetadata(command.Flags(), &opts.userMetadata, cmd.PflagUserMetadataVerifyUsage)
+	command.MarkFlagRequired("signature")
+	return command
+}
+
+func runVerify(command *cobra.Command, cmdOpts *blobVerifyOpts) error {
+	// set log level
+	ctx := cmdOpts.LoggingFlagOpts.InitializeLogger(command.Context())
+
+	// initialize
+	displayHandler := display.NewBlobVerifyHandler(cmdOpts.Printer)
+	blobFile, err := os.Open(cmdOpts.blobPath)
+	if err != nil {
+		return err
+	}
+	defer blobFile.Close()
+
+	signatureBytes, err := os.ReadFile(cmdOpts.signaturePath)
+	if err != nil {
+		return err
+	}
+	blobVerifier, err := cmd.GetVerifier(ctx, true)
+	if err != nil {
+		return err
+	}
+
+	// set up verification plugin config
+	pluginConfigs, err := cmd.ParseFlagMap(cmdOpts.pluginConfig, cmd.PflagPluginConfig.Name)
+	if err != nil {
+		return err
+	}
+
+	// set up user metadata
+	userMetadata, err := cmd.ParseFlagMap(cmdOpts.userMetadata, cmd.PflagUserMetadata.Name)
+	if err != nil {
+		return err
+	}
+	signatureMediaType, err := parseSignatureMediaType(cmdOpts.signaturePath)
+	if err != nil {
+		return err
+	}
+	verifyBlobOpts := notation.VerifyBlobOptions{
+		BlobVerifierVerifyOptions: notation.BlobVerifierVerifyOptions{
+			SignatureMediaType: signatureMediaType,
+			PluginConfig:       pluginConfigs,
+			UserMetadata:       userMetadata,
+			TrustPolicyName:    cmdOpts.policyStatementName,
+		},
+		ContentMediaType: cmdOpts.blobMediaType,
+	}
+	_, outcome, err := notation.VerifyBlob(ctx, blobVerifier, blobFile, signatureBytes, verifyBlobOpts)
+	outcomes := []*notation.VerificationOutcome{outcome}
+	err = ioutil.PrintVerificationFailure(outcomes, cmdOpts.blobPath, err, true)
+	if err != nil {
+		return err
+	}
+	displayHandler.OnVerifySucceeded(outcomes, cmdOpts.blobPath)
+	return displayHandler.Render()
+}
+
+// parseSignatureMediaType returns the media type of the signature file.
+// `application/jose+json` and `application/cose` are supported.
+func parseSignatureMediaType(signaturePath string) (string, error) {
```

Comment:
Just a note that this method is also implemented in #1137. Duplicated code should be avoided.